app/src/main/java/com/sopt/yeogieottae/ui/compare/CompareViewModel.kt

```diff
@@ -3,87 +3,30 @@ package com.sopt.yeogieottae.ui.compare
-                success = true,
-                message = "비교하기 목록 불러오기를 성공했습니다.",
-                data = emptyList()
-            )
-        } else {
-            ApiResponse(
-                code = 200,
-                success = true,
-                message = "비교하기 목록 불러오기를 성공했습니다.",
-                data = listOf(
-                    Room(
-                        roomId = 1,
-                        hotelName = "그랜드 인터컨티넨탈 파르나스",
-                        roomType = "클래식 킹",
-                        price = 464640,
-                        reviewRate = 9.4,
-                        reviewCount = 2183,
-                        imageUrl = "image1"
-                    ),
-                    Room(
-                        roomId = 2,
-                        hotelName = "그랜드 인터컨티넨탈 파르나스",
-                        roomType = "주니어 스위트 킹",
-                        price = 608679,
-                        reviewRate = 9.4,
-                        reviewCount = 2183,
-                        imageUrl = "image2"
-                    ),
-                    Room(
-                        roomId = 3,
-                        hotelName = "서울 신라 호텔",
-                        roomType = "디럭스 트윈",
-                        price = 497000,
-                        reviewRate = 9.8,
-                        reviewCount = 1183,
-                        imageUrl = "image3"
-                    ),
-                    Room(
-                        roomId = 4,
-                        hotelName = "서울 신라 호텔",
-                        roomType = "디럭스 트윈",
-                        price = 497000,
-                        reviewRate = 9.8,
-                        reviewCount = 1183,
-                        imageUrl = "image3"
-                    ),
-                    Room(
-                        roomId = 5,
-                        hotelName = "서울 신라 호텔",
-                        roomType = "디럭스 트윈",
-                        price = 497000,
-                        reviewRate = 9.8,
-                        reviewCount = 1183,
-                        imageUrl = "image3"
-                    ),
-                    Room(
-                        roomId = 6,
-                        hotelName = "서울 신라 호텔",
-                        roomType = "디럭스 트윈",
-                        price = 497000,
-                        reviewRate = 9.8,
-                        reviewCount = 1183,
-                        imageUrl = "image3"
-                    )
-                )
-            )
+    fun fetchCompareData() {
+        viewModelScope.launch {
+            try {
+                val response = ServicePool.authService.compare()
+                if (response.isSuccessful) {
+                    _compareResponse.value = response.body()
+                } else {
+                    _message.value = "CompareViewModel 응답 실패"
+                }
+            } catch (e: Exception) {
+                _message.value = "CompareViewModel 에러"
```

Comment:
```suggestion
               runCatching {
                ServicePool.authService.compare()
            }.onSuccess { response ->
                _compareResponse.value = response.body()
            }.onFailure { e ->
                if(e is HttpException)  _message.value = "CompareViewModel 응답 실패"
                else _message.value = "CompareViewModel 에러"
            }
        }
```

이렇게 runcatch 문으로 바꿀 수 있을 거 같아요 
  if(e is HttpException)  _message.value = "CompareViewModel 응답 실패" 의 경우 서버 통신 과정에서 문제가 생겼을 때,


else 의 경우 이외의 예외 사항에 대해서 적용이 됩니다.